spec/requests/api/v2/session_spec.rb

```diff
@@ -0,0 +1,88 @@
+require "rails_helper"
+
+describe '/api/v2/session', type: :request do
+  let(:user)    { create(:user) }
+  let(:payload) { { email: user.email } }
+  let(:token) do
+    JWT.encode payload.to_json,
+               Settings.authentication.secret,
+               Settings.authentication.algorithm
+  end
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  context 'when unauthenticated' do
+    let(:payload) { { email: 'foo@bar' } }
+
+    before do
+      get "/api/v2/users/#{user.id}",
+          headers: { 'HTTP_AUTHORIZATION' => credentials }
+    end
+
+    subject { response }
+
+    it { should have_http_status(:unauthorized) }
+  end
+
+  describe 'creating a session' do
+    let(:user)    { create(:user, last_login_date_utc: 10.days.ago) }
+      # macOS is in not defintive in accuracy as its lacks nano seconds vs other OS
+      # TimeCop therefore has slight flaw in design
+    it 'saves the last login time' do
+      Timecop.freeze do
+        post '/api/v2/session',
+             headers: { 'HTTP_AUTHORIZATION' => credentials }
+
+        expect(user.reload.last_login_date_utc).to be_within(1.second).of Time.now.utc
```

Comment:
Not quite sure what your above comment about Timecop is saying, but since you're using it you can simplify this test to be:

```
expect(user.reload.last_login_date_utc).to eq Time.now.utc
```

Reply:
breeze thro this

https://github.com/travisjeffery/timecop/issues/97

Reply:
```
1) /api/v2/session creating a session saves the last login time
     Failure/Error: expect(user.reload.last_login_date_utc).to eq Time.now.utc
     
       expected: 2019-02-26 12:52:26.530098940 +0000
            got: 2019-02-26 12:52:26.530098000 +0000
```

# macOS is in not defintive in accuracy as its lacks nano seconds vs other OS
# TimeCop therefore has slight flaw in design

I use linux vs you use mac.
TimeCop was most likely tied to mac defects.

Unless i wrapped it in wrong block...

Reply:
Now I'm just going to throw db in the equation as well.

Reply:
But, if I change this test to test for equality (see my original comment), it works (tested locally). Have you tried it and had it fail?

Reply:
Ok, I'm starting to understand what you mean Sau.

On Linux this test fails because the nanoseconds are different (Timecop hasn't frozen that, it seems). On OSX it doesn't fail, either because the nanoseconds are the same or maybe because `eq` doesn't compare them.

Reply:
To be correct in case anyone else wants to know

On a mac it nanosecond is zeroed out
On database when retrieving a date column nanosecond is not save, there for nanosecond is zeroed out.

Timecop do free the time inclusive of nanoseconds, but there seems to be other issues that question the overall functionality concerning time.